src/main/java/org/springframework/data/r2dbc/repository/query/ConditionFactory.java

```diff
@@ -0,0 +1,307 @@
+import org.springframework.data.mapping.PropertyPath;
+import org.springframework.data.r2dbc.convert.R2dbcConverter;
+import org.springframework.data.r2dbc.core.ReactiveDataAccessStrategy;
+import org.springframework.data.relational.core.mapping.RelationalPersistentEntity;
+import org.springframework.data.relational.core.mapping.RelationalPersistentProperty;
+import org.springframework.data.relational.core.sql.*;
+import org.springframework.data.relational.core.sql.render.RenderNamingStrategy;
+import org.springframework.data.repository.query.parser.Part;
+import org.springframework.util.Assert;
+
+/**
+ * Simple factory to contain logic to create {@link Condition}s from {@link Part}s.
+ *
+ * @author Roman Chigvintsev
+ */
+class ConditionFactory {
+    private final ReactiveDataAccessStrategy dataAccessStrategy;
+    private final RenderNamingStrategy namingStrategy;
+    private final ParameterMetadataProvider parameterMetadataProvider;
+
+    /**
+     * Creates new instance of this class with the given {@link ReactiveDataAccessStrategy},
+     * {@link RenderNamingStrategy} and {@link ParameterMetadataProvider}.
+     *
+     * @param dataAccessStrategy        data access strategy (must not be {@literal null})
+     * @param namingStrategy            naming strategy for SQL rendering (must not be {@literal null})
+     * @param parameterMetadataProvider parameter metadata provider (must not be {@literal null})
+     */
+    ConditionFactory(ReactiveDataAccessStrategy dataAccessStrategy,
+                     RenderNamingStrategy namingStrategy,
+                     ParameterMetadataProvider parameterMetadataProvider) {
+        Assert.notNull(dataAccessStrategy, "Reactive data access strategy must not be null!");
+        Assert.notNull(namingStrategy, "Render naming strategy must not be null!");
+        Assert.notNull(parameterMetadataProvider, "Parameter metadata provider must not be null!");
+
+        this.dataAccessStrategy = dataAccessStrategy;
+        this.namingStrategy = namingStrategy;
+        this.parameterMetadataProvider = parameterMetadataProvider;
+    }
+
+    /**
+     * Creates condition for the given {@link Part}.
+     *
+     * @param part method name part (must not be {@literal null})
+     * @return {@link Condition} instance
+     * @throws IllegalArgumentException if part type is not supported
+     */
+    public Condition createCondition(Part part) {
+        Part.Type type = part.getType();
+        switch (type) {
+            case BETWEEN: {
+                Expression pathExpression = createPropertyPathExpression(part.getProperty());
+                BindMarker firstBindMarker = createBindMarker(parameterMetadataProvider.next(part));
+                BindMarker secondBindMarker = createBindMarker(parameterMetadataProvider.next(part));
+
+                // TODO: why do not we have BETWEEN condition?
+                return Conditions.isGreaterOrEqualTo(pathExpression, firstBindMarker)
```

Comment:
Good candidate for an enhancement in Spring Data Relational. On a related note: We support in some stores also a `Range` data type (`findByDateBetween(Range<LocalDate>)`) that allows for more fine-grained control of inclusion/exclusion regarding boundaries.